### PR TITLE
Vs 1063 bit pack ref ranges data into a more efficient representation

### PR DESCRIPTION
**Pull request by koncheto-broad**
Adding in the option for using a compressed representation of our reference data.  The flag use_compressed_references needs to be passed into ingest and also export, and doing so causes us to use a different internal schema for the ref_ranges table that can save ~40% on space.  On the VCF path, the packed data is expanded while we populate the prepare tables before extract.  On the VDS path, the packed data is expanded while we extract from the tables to create the Avro files.

I updated integration tests to take use_compressed_refs as an option, and saw what I needed.  Integrations tests for the VCF and VDS paths go to completion and fail, but in the expected way. AssertIdenticalOutputs, the important part, passes.  AssertTableSizesAreExpected fails because the ref_ranges table is so much smaller (255884464 vs 431805033 expected) and AssertCostIsTrackedAndExpected fails because the PrepareRangesCallsetTask.GvsPrepareRanges.BigQuery Query Scanned is similarly smaller (340787200 vs 515899392).  Given the nature of the change and the fact that it is an optional flag that defaults to zero, updating the integration tests to cover this path did not seem sound at this time.  But using them to verify that the _content_ of the final vcfs wasn't affected by the change makes more sense.

VCF integration run: https://job-manager.dsde-prod.broadinstitute.org/jobs/e9896786-9d3f-48b9-ab6f-9a76d9fafafe
Hail integration run: https://job-manager.dsde-prod.broadinstitute.org/jobs/dbdd0934-50c3-4477-a191-3282341eacd3

another post-merge integration run with same results: identical outputs, failure due to lower table sizes and query scan costs
https://job-manager.dsde-prod.broadinstitute.org/jobs/edabdac3-1856-4e2a-aadc-3bcd4c353956

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java
Comment (mcovarr) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java:89]: OCD: If the high bit of `chromosomeIndex` was set (not that it theoretically could be) wouldn't this overflow a signed `long`? Maybe a `0x7FFF` mask?

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java
Comment (mcovarr) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java:96]: There are a lot of "not covered by tests" warnings on the logic above that looks like it really could / should be covered by unit tests.

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java
Comment (mcovarr) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java:91]: are we very sure there are no indels longer than this?

File: scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl:296]: Something I think I've heard @gbggrant ask for: can we have these functions made available in the GVS schema so ad-hoc querying does not become unpossible (or very unconvenient)?

**Review by mcovarr — CHANGES_REQUESTED** [review]
Needs documentation, unit tests over core logic, WDL fix.

File: scripts/variantstore/wdl/GvsBulkIngestGenomes.wdl
Comment (gbggrant) [reply scripts/variantstore/wdl/GvsBulkIngestGenomes.wdl:136]: Yeah, thought the womtools check on the PR would have failed.

File: scripts/variantstore/wdl/GvsUtils.wdl
Comment (gbggrant) [inline scripts/variantstore/wdl/GvsUtils.wdl:74]: Will need to be updated with the changes I merged to `ah_var_store` today.

**Review by gbggrant — COMMENTED** [review]
Looks good. A couple of comments.
Also, please put in the code / documentation somewhere the representation of the packed ref_ranges table (the one in the spike: https://broadworkbench.atlassian.net/browse/VS-973

File: scripts/variantstore/wdl/GvsBulkIngestGenomes.wdl
Comment (koncheto-broad) [reply scripts/variantstore/wdl/GvsBulkIngestGenomes.wdl:136]: and yet it does actually work, weirdly enough.  I'll happily slip the comma in, though.

File: scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl
Comment (koncheto-broad) [reply scripts/variantstore/wdl/GvsExtractAvroFilesForHail.wdl:296]: not in the first pass, although the SQL to do the queries are present in the tickets including examples of how to use them to make the content look like before.  I am MUCH more interested in getting the initial implementation in, given our timelines, than minimizing the developer effort.  In the meantime, when I do ad-hoc querying on tables, I just copy/paste the file of utility functions into my sql console and use them

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java
Comment (koncheto-broad) [reply src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java:89]: Not an incorrect observation.  But when choosing the number of bits to allocate, I did some googling.  Apparently the highest number of chromosomes ever discovered in an organism (according to Guinness World Record) is some random single-celled thing that has 15.6k.  The number of allocated bits allows for over 65k.  We have a LARGE buffer before we'd ever encounter any issues with any undiscovered organisms that could come close to overflowing the allocated bits.  It's a theoretical possibility that is impossible to actually encounter

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java
Comment (koncheto-broad) [reply src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java:91]: This length does not refer to the length of a variant, but the length of a _reference block_.  These are capped at 1k in our code, and the number of bits allocated would allow us to expand that up to 4k without having to adjust the representation

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/ingest/CreateVariantIngestFiles.java
Comment (koncheto-broad) [reply src/main/java/org/broadinstitute/hellbender/tools/gvs/ingest/CreateVariantIngestFiles.java:165]: 40% logical

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/ingest/RefRangesAvroWriter.java
Comment (koncheto-broad) [reply src/main/java/org/broadinstitute/hellbender/tools/gvs/ingest/RefRangesAvroWriter.java:54]: Yeah, it's beyond the scope of this ticket. But I agree that there are a lot of historical artifacts, especially through our ingest code that we can clear out and simplify.  Case in point: I had to make changes to what I am 99% sure is dead code in order to make this change.  NOT fun.

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java
Comment (koncheto-broad) [reply src/main/java/org/broadinstitute/hellbender/tools/gvs/common/SchemaUtils.java:96]: Added a unit testes for the new code I put in SchemaUtils

File: scripts/variantstore/wdl/GvsUtils.wdl
Comment (koncheto-broad) [reply scripts/variantstore/wdl/GvsUtils.wdl]: merged in ah_var_store and made new dockers yesterday

File: scripts/variantstore/wdl/GvsImportGenomes.wdl
Comment (gbggrant) [inline scripts/variantstore/wdl/GvsImportGenomes.wdl:222]: Why is it defaulted to true here?

File: scripts/variantstore/wdl/GvsUtils.wdl
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsUtils.wdl:75]: just sanity checking that you and @gbggrant are not going to collide since we're still living in a hashless world until [this gets merged](https://github.com/broadinstitute/gatk/pull/8549)

**Review by mcovarr — APPROVED** [review]
LGTM if we're very sure there won't be Docker collisions

**Review by gbggrant — APPROVED** [review]

File: scripts/variantstore/wdl/GvsImportGenomes.wdl
Comment (koncheto-broad) [reply scripts/variantstore/wdl/GvsImportGenomes.wdl:222]: Initial, small-scale testing.  I thought I swept over the code and changed all arguments to be false by default, but I must've missed this one.  Nice catch!  I'll change it now

File: scripts/variantstore/wdl/GvsUtils.wdl
Comment (koncheto-broad) [reply scripts/variantstore/wdl/GvsUtils.wdl:75]: I pulled in ah_var_store again, saw george's change to both of those, redid both of mine, and then ran an integration test again afterwards.  We should be fine

**Pull request merged**
